src/packageurl/contrib/purl2url.py

```diff
@@ -75,7 +76,12 @@ def get_download_url(purl):
         return download_url
 
     # Fallback on the `download_url` qualifier when available.
-    purl_data = PackageURL.from_string(purl)
+    purl_data = None
```

Comment:
I am not sure we want to swallow an exception here. If the purl is invalid, let the error bubble up and do not catch the exception.

Reply:
Thanks @pombredanne -- I have updated `get_download_url()` to the following (actually, this reverts the code to its state before I added a try/except):

```
def get_download_url(purl):
    """
    Return a download URL inferred from the `purl` string.
    """
    download_url = _get_url_from_router(download_router, purl)
    if download_url:
        return download_url

    # Fallback on the `download_url` qualifier when available.
    purl_data = PackageURL.from_string(purl)
    return purl_data.qualifiers.get("download_url", None)
```

I also removed `"pkg:cocoapods/": [],` from `test_purl2url_get_inferred_urls()` because otherwise the test fails w/o the try/except I've removed.